### ISSUE TITLE
Document MySQL Permissions required for Sqitch registry creation

### ISSUE THREAD

**Issue by jddulin** [sqitchers/sqitch#488]
First of all, thanks to the contributors of Sqitch for all of your work!  It's a great, simple tool.

I am struggling to run my first deployment of a newly initialized Sqitch project.  I have successfully written and tested my `deploy`, `revert`, and `verify` code on a local test DB.  However, when I run the deployment on our prod machine, I consistently get an error,
```
access denied for user 'john_dulin on database 'sqitch'
```

When my DBA updates my permissions on the newly created registry DB, and I try to re-rerun
```
sqitch deploy prod --verbose
```
we get an error saying that the "'releases' table already exists", implying the migration is trying to re-run the registry creation from the beginning and failing.

So the only way we found to get around this issue was to give me root permissions - Then, `sqitch deploy` runs beautifully after blowing away the messy registry.  Is there an enumeration of the permissions required for creating the registry table somewhere, and if not, could we document that? I thought it would be simple, such as 'CREATE DATABASE, CREATE TABLE, INSERT, etc.' but there appears to be some missing permissions I do not inherit to the newly created `sqitch` table w/o root.  This would make allowing different users or service accounts run their own migrations easier - Thank you! 

versions,
```
sqitch (App::Sqitch) v1.0.0
Server version: 5.6.10 MySQL Community Server (GPL)
```

**Comment by theory** [comment]
The registry is created as its own MySQL database on your first `deploy`, so it needs the ability to create objects from that point, at least. I don't know which permissions, exactly, not being a MySQL expert, but it runs [this script](/sqitchers/sqitch/blob/develop/lib/App/Sqitch/Engine/mysql.sql) through the `mysql` client to create the database and its objects, so maybe read through that to see what permissions might be missing? Happy to add the list of the appropriate permissions to the tutorial.

**Comment by theory** [comment]
Hi @jddulin. Were you able to get Sqitch to create the registry database on your MySQL server?

**Comment by theory** [comment]
> **Note:** I'm aware that Sqitch officially supports ClickHouse version 25.8 and above. I was testing with ClickHouse 25.6, which is below the minimum requirement.

Yes, if you don't revert it should be fine.

> ### Issue 1: Registry Database Creation Fails Despite Sufficient Permissions
> 
> During `sqitch deploy`, the Sqitch registry database was not created automatically. It failed with a permission error, even though the user had all the required access permissions granted:

Did you actually have a host name in the URI? If not, I believe that issue was fixed in 4f04e7b. The issue is that it was connecting to the wrong database (`default`) for the registry if no host name was set.

> ### Issue 2: Sqitch Attempts to Recreate Existing Registry Tables
> 
> During `sqitch deploy`, the tool sometimes attempts to recreate all the Sqitch registry tables, even when they already exist. This causes deployment to fail:

This would be the same issue if you used no host name.

> Could we add `IF NOT EXISTS` clauses to the ClickHouse Sqitch registry DDL commands in the [clickhouse.sql file](https://github.com/sqitchers/sqitch/blob/develop/lib/App/Sqitch/Engine/clickhouse.sql)? This would make the registry creation idempotent and prevent failures when tables already exist.

It should never reach that point. However, the ClickHouse ODBC driver has a bug where it returns the same error code for every error it encounters. If it gets any error attempting to connect to the registry database, it assumes it's a missing table error (after attempting to query `changes`) and goes to try to create the registry, assuming there is none.
